### ISSUE TITLE
Add time out parameter 

### ISSUE THREAD

**Issue by FMCumhaill** [dmarx/psaw#20]
Currently using Signal but a "Max search time" would be super handy! 

Love the library tho!

**Comment by dmarx** [comment]
What's Signal?

I'm sort of disinclined to add this since the best way to implement it I can think of would be equivalent to

```
start=time.time()
for rec in gen:
    if start - time.time() > max_search_time:
        break
```

The issue is that I don't know how to implement something like this if the search is hanging because pushshift is taking a long time to deliver results, or how backoff/retries should be incorporated here.

I think the main reason you want this is because psaw sometimes gets caught in a loop where it's not actually getting new items but just requesting the same 500 results repeatedly. I'm not sure if this has been completely squashed, but at least the main culprit should be taken care of in the latest release. 

Although I don't think this is something I'll add, I'm going to leave this open to invite other people to chime in if this is something lots of people would like, and maybe someone has a more elegant implementation idea.

**Comment by dmarx** [comment]
uh... oh, [signal](https://docs.python.org/2/library/signal.html), got it. That looks useful and like it solves the issue I described above. Seeing as I've never used this library and you already have a working solution for your use case, wanna submit a pull request or maybe post a demo of how you're using it with psaw?

**Comment by FMCumhaill** [comment]
Yes Signal sorry >.<, and yes that's the issue I was trying to fix. Unfortunately signal only works for Linux/Max machines. 

In the end I couldn't actually get it working if I remember correctly. You'd probably want to write in a [multiprocessing aspect to the function call](https://stackoverflow.com/questions/492519/timeout-on-a-function-call). I really hate coding multiprocessing functions, but if I get time (not likely) i'll look at it. Otherwise let's leave this OPEN to see if anyone else jumps in?
